march_hardware/src/joint.cpp

```diff
@@ -131,6 +131,15 @@ double Joint::getVelocity() const
   return this->velocity_;
 }
 
+double Joint::getVoltageVelocity() const
+{
+  double R_motor = 0.05;
+  double Kv_motor = 355;
+  double rpmToRad = M_PI / 30;
+  double Ke = Kv_motor * rpmToRad;
+  return (this->imc_->getMotorVoltage() + this->imc_->getMotorCurrent() * R_motor) / Ke;
```

Comment:
```suggestion
  double resistance_motor = 0.05;
  double Kv_motor = 355;
  double rpm_to_rad = M_PI / 30;
  double Ke = Kv_motor * rpm_to_rad;
  return (this->imc_->getMotorVoltage() + this->imc_->getMotorCurrent() * resistance_motor) / Ke;
```
Non constant variables are usually named with non-capitals and underscores ;) what are Ke and Kv? maybe write out the abbreviations?

Reply:
@Roelemans https://en.wikipedia.org/wiki/Motor_constants#Motor_velocity_constant,_back_EMF_constant This is the information about the Kv and Ke value. The reason I put both of them in there is because we got the Kv_motor directly from the spec sheet of the motor and I need Ke in the derivation in line 140. We can change it to backemf_motor_constant, but I am not sure if this is the better way. Maybe stay with the Kv and Ke and add the wikipedia link?

Reply:
About 'non constant variables', I wonder whether the first four variables are constants. Because they are pretty constant, right?

Reply:
Yes they are very constant. According to [ROS C++ style guide](https://wiki.ros.org/CppStyleGuide#Constants) they should be CAPITAL_SNAKE_CASE. However, since we only do this for global constants I think they should be normal snake_case

```suggestion
  // https://en.wikipedia.org/wiki/Motor_constants#Motor_velocity_constant,_back_EMF_constant
  const double resistance = 0.05;
  const double velocity_constant = 355;
  const double rpm_to_rad = M_PI / 30;
  const double electric_constant = velocity_constant * rpm_to_rad;
  return (this->imc_->getMotorVoltage() + this->imc_->getMotorCurrent() * resistance) / electric_constant;
```
Personally I would do it like this, but you can also choose to keep the `kv` and `ke` names, since those better reflect the theory. Most important is **keep it consistent**. You know variable naming is one of the hardest things in programming :wink:

Normally hardcoding constants is not approved, since these values depend on the motor you are using. Perhaps we should add motor parameters to the robot yaml, but that is something for another time.